### ISSUE TITLE
when convert StridedSlice, error return when optimizing gpu memory.

### ISSUE THREAD

**Issue by huanyingjun** [XiaoMi/mace#122]
my model is TensorFlow, and has StridedSlice op, when convert into mace GPU, shows below log, 
```
Traceback (most recent call last):
File "/home/liming/work/ori_mace/mace/bazel-bin/mace/python/tools/converter.runfiles/mace/mace/python/tools/converter.py", line 298, in 
main(unused_args=[sys.argv[0]] + unparsed)
File "/home/liming/work/ori_mace/mace/bazel-bin/mace/python/tools/converter.runfiles/mace/mace/python/tools/converter.py", line 185, in main
memory_optimizer.optimize_gpu_memory(output_graph_def)
File "/home/liming/work/ori_mace/mace/bazel-bin/mace/python/tools/converter.runfiles/mace/mace/python/tools/memory_optimizer.py", line 244, in optimize_gpu_memory
mem_optimizer.optimize()
File "/home/liming/work/ori_mace/mace/bazel-bin/mace/python/tools/converter.runfiles/mace/mace/python/tools/memory_optimizer.py", line 113, in optimize
op.output_shape[i].dims)
File "/home/liming/work/ori_mace/mace/bazel-bin/mace/python/tools/converter.runfiles/mace/mace/python/tools/memory_optimizer.py", line 217, in get_op_mem_block
raise Exception('output shape dim size is not 2 or 4.')
Exception: output shape dim size is not 2 or 4.
Traceback (most recent call last):
File "tools/converter.py", line 1595, in 
flags.func(flags)
File "tools/converter.py", line 785, in convert_func
convert_model(configs)
File "tools/converter.py", line 711, in convert_model
",".join(model_config.get(YAMLKeyword.graph_optimize_options, [])))
File "/home/liming/work/ori_mace/mace/tools/sh_commands.py", line 524, in gen_model_code
_fg=True)
File "/usr/local/lib/python2.7/dist-packages/sh.py", line 1413, in call
raise exc
sh.ErrorReturnCode_1:
```

and I add some log in memory_optimizer.py , found that the output shape length of StridedSlice  is 0

```py
    def get_op_mem_block(self, op_type, output_shape):
        mem_block = [0, 0]
        print (op_type)  #====> StridedSlice 
        print (len(output_shape))  #=====> 0
        if op_type == 'WinogradTransform' or op_type == 'MatMul':
            mem_block[0] = output_shape[2]
            mem_block[1] = output_shape[0] * int((output_shape[1] + 3) / 4)
        elif op_type == 'Shape':
            mem_block[0] = output_shape[0]
            mem_block[1] = 1
        else:
            if len(output_shape) == 2:  # only support fc/softmax
                mem_block[0] = int((output_shape[1] + 3) / 4)
                mem_block[1] = output_shape[0]
            elif len(output_shape) == 4:
                mem_block[0] = output_shape[2] * int((output_shape[3] + 3) / 4)
                mem_block[1] = output_shape[0] * output_shape[1]
            else:
                raise Exception('output shape dim size is not 2 or 4.')
        return mem_block
```

**Comment by llhe** [comment]
Can you confirm why the shape is 0? It's undesirable.

**Comment by cutejerry** [comment]
@llhe 
I think maybe the shape of strided_slice is undetermined. It's according to input.
We checked the tensorflow tensor:
```shell
[<tf.Tensor 'net/flatten/strided_slice:0' shape=() dtype=int32>]
```
It seems there is no shape information also.

And the converter won't update output_shape (empty)
```python
    def convert_general_op(self, tf_op):
        op = self._mace_net_def.op.add()
        op.name = tf_op.name
        op.type = tf_op.type
        op.input.extend([tf_input.name for tf_input in tf_op.inputs])
        op.output.extend([tf_output.name for tf_output in tf_op.outputs])
        for tf_output in tf_op.outputs:
            output_shape = op.output_shape.add()
            output_shape.dims.extend(self.infer_tensor_shape(tf_output))
```

**Comment by llhe** [comment]
@TonyMou Can you have a look at this?

**Comment by TonyMou** [comment]
@huanyingjun StridedSlice does not support scalar output in MACE. This is a bug and will be fixed soon.

**Comment by chuan92** [comment]
I have met the same problem when using conv2d_transpose which called strided_slice  op.

**Comment by chuan92** [comment]
any update on this issue?

**Comment by liyinhgqw** [comment]
@TonyMou Could you take a look at this issue?

**Comment by TonyMou** [comment]
Still working on it.

**Comment by akbarxie** [comment]
这个问题解决了吗？

**Comment by TonyMou** [comment]
已经解决了，代码这两天就会更新上去。



发自我的小米手机
在 akbarxie <notifications@github.com>，2018年8月21日 下午3:11写道：

这个问题解决了吗？

**Comment by terryzhangthu** [comment]
I met same problem when convert the tensorflow conv2d_tranpose, is this issue solved? 
could you let me know the commit ID for this if solution is already updated?
Following is my converter log...
//////////////
conv2d/Conv2D (Conv2D): [[1L, 1L, 128L, 128L]]
conv2d_transpose/strided_slice (StridedSlice): [[]]
conv2d_transpose/strided_slice_1 (StridedSlice): [[]]
conv2d_transpose/mul (Eltwise): [[]]
conv2d_transpose/strided_slice_2 (StridedSlice): [[]]
conv2d_transpose/mul_1 (Eltwise): [[]]
///////////////////

**Comment by TonyMou** [comment]
already resolved in commit 32be5ec866e760aac1b8235d75a3f581ea62b95c

**Comment by yk0607** [comment]
这个问题有方法解决吗  ？    
模型输出为 3 dim 时 总会报这样错，  是不支持 3 dim 输出吗？
